### ISSUE TITLE
Syntax of complete.sh is wrong, breaks SSH connections

### ISSUE THREAD

**Issue by cpitclaudel** [ocaml/opam#2899]
OPAM's `complete.sh` script uses `bash`-specific syntax, but runs even when the shell is `sh`.  This breaks e.g. Tramp, which runs more or less `ssh remote-server "/bin/sh -l -c 'echo $PATH'; echo exit_status: $?"` and gets an exit code of `2`.

And indeed, once one removes the many layer of redirection to `/dev/null`, one gets this:

```
sh: 42: ./complete.sh: Syntax error: "(" unexpected (expecting "}")
```

Shouldn't OPAM's scripts either be written in SH syntax, or only be sourced when the shell is bash?

**Comment by AltGr** [comment]
`complete.sh` is clearly intended as bash-specific, and shouldn't be loaded if the shell is `sh`.
We could wrap the complete script with a check of the environment to detect the running shell; but the current call should actually not be causing trouble:
```
if tty -s >/dev/null 2>&1; then
  test -r /home/lg/.opam/opam-init/complete.sh && . /home/lg/.opam/opam-init/complete.sh > /dev/null 2> /dev/null || true
fi
```

**Comment by cpitclaudel** [comment]
Indeed, I noticed that there's a multiple levels of `|| true` in there — but commenting out the line added by OPAM to my `.profile` fixed the exit code.

**Comment by AltGr** [comment]
Probably `sh` can't backtrack to the file above when including with `. file` on a syntax error: inclusion might be syntactic. But besides that, it should only get sourced when running from a tty, which shouldn't have been your case ?

**Comment by cpitclaudel** [comment]
> But besides that, it should only get sourced when running from a tty, which shouldn't have been your case ?

No: Tramp (Emacs' remote editing package) uses a TTY by default (with `TERM=dumb`)

**Comment by AltGr** [comment]
Ah, I see; will add a safeguard to the completion scripts then
